sympy/ntheory/residue_ntheory.py

```diff
@@ -433,10 +433,34 @@ def _sqrt_mod_tonelli_shanks(a, p):
     """
     Returns the square root in the case of ``p`` prime with ``p == 1 (mod 8)``
 
+    Assume that the root exists (if it does not, we are in an infinite loop).
+    Although ``p`` correctly returns the answer for any odd prime,
+    ``p != 1 (mod 8)``, there is no advantage to using this algorithm since
+    a more efficient algorithm exists.
+
+    Parameters
+    ==========
+
+    a : integer
+    p : odd prime number
```

Comment:
```suggestion
    a : int
    p : int
        Odd prime number
```

I now believe that with the sphinx able to pull out the type annotations, it's better to make the type annotation actual python type annotations, and move semantics to details